### PR TITLE
api key for api and hash support

### PR DESCRIPTION
**Pull request by taylorking**

### CODE REVIEW

**Comment by taylorking** [comment]
fixes #76 #85 #95 work around for #74

File: api-gateway-config/scripts/lua/policies/security/apiKey.lua [hunk 1]
```diff
@@ -40,9 +40,8 @@ local _M = {}
 -- @param scope scope of the subscription
 -- @param apiKey the subscription api key
 -- @param return boolean value indicating if the subscription exists in redis
-function validate(tenant, gatewayPath, apiId, scope, apiKey)
+function validate(red, tenant, gatewayPath, apiId, scope, header, apiKey)
```
Comment (alexsong93) [inline api-gateway-config/scripts/lua/policies/security/apiKey.lua]: Can you add what the additional parameters are to the doc above?

File: api-gateway-config/scripts/lua/policies/security/apiKey.lua [hunk 2]
```diff
@@ -51,31 +50,53 @@ function validate(tenant, gatewayPath, apiId, scope, apiKey)
...
+  local result = processWithRedis(red, securityObj, sha256)
+  redis.close(red)
+  return result
+end 
+
 --- Process the security object
 -- @param securityObj security object from nginx conf file
 -- @return apiKey api key for the subscription
-function process(securityObj)
+function processWithRedis(red, securityObj, hashFunction)
```
Comment (alexsong93) [inline api-gateway-config/scripts/lua/policies/security/apiKey.lua]: Same comment as above

File: api-gateway-config/scripts/lua/policies/security/apiKey.lua [hunk 3]
```diff
@@ -51,31 +50,53 @@ function validate(tenant, gatewayPath, apiId, scope, apiKey)
...
   local tenant = ngx.var.tenant
   local gatewayPath = ngx.var.gatewayPath
-  local apiId = ngx.var.apiId
+  local apiId = redis.resourceToApi(red, utils.concatStrings({'resources:', tenant, ':', gatewayPath})) 
+  -- local apiId = ngx.var.apiI
```
Comment (alexsong93) [inline api-gateway-config/scripts/lua/policies/security/apiKey.lua]: remove comment here

**Pull request merged**